diff -bur tcl8.6.13.bak/unix/Makefile.in tcl8.6.13/unix/Makefile.in
--- tcl8.6.13.bak/unix/Makefile.in	2022-10-28 12:19:48.000000000 -0300
+++ tcl8.6.13/unix/Makefile.in	2024-01-04 17:30:17.052772000 -0400
@@ -1072,7 +1072,7 @@
 	@for i in $(GENERIC_DIR)/tclInt.h $(GENERIC_DIR)/tclIntDecls.h \
 		$(GENERIC_DIR)/tclIntPlatDecls.h $(GENERIC_DIR)/tclPort.h \
 		$(GENERIC_DIR)/tclOOInt.h $(GENERIC_DIR)/tclOOIntDecls.h \
-		$(UNIX_DIR)/tclUnixPort.h; \
+		$(UNIX_DIR)/tclUnixPort.h $(GENERIC_DIR)/tclCompile.h; \
 	    do \
 	    $(INSTALL_DATA) $$i "$(PRIVATE_INCLUDE_INSTALL_DIR)"; \
 	done
diff -bur tcl8.6.13.bak/win/Makefile.in tcl8.6.13/win/Makefile.in
--- tcl8.6.13.bak/win/Makefile.in	2022-10-28 12:19:48.000000000 -0300
+++ tcl8.6.13/win/Makefile.in	2024-01-04 17:28:42.340922600 -0400
@@ -781,7 +781,7 @@
 	@for i in "$(GENERIC_DIR)/tclInt.h" "$(GENERIC_DIR)/tclIntDecls.h" \
 		"$(GENERIC_DIR)/tclIntPlatDecls.h" "$(GENERIC_DIR)/tclPort.h" \
 		"$(GENERIC_DIR)/tclOOInt.h" "$(GENERIC_DIR)/tclOOIntDecls.h" \
-		"$(WIN_DIR)/tclWinPort.h" ; \
+		"$(WIN_DIR)/tclWinPort.h" "$(GENERIC_DIR)/tclCompile.h"; \
 	    do \
 	    $(COPY) "$$i" "$(PRIVATE_INCLUDE_INSTALL_DIR)"; \
 	    done;
