diff -ruN A/pkgIndex.tcl.in B/pkgIndex.tcl.in
--- A/pkgIndex.tcl.in	2024-07-19 23:09:24.599474200 -0300
+++ B/pkgIndex.tcl.in	2026-04-04 00:00:00.000000000 -0300
@@ -4,7 +4,13 @@
 if {[package vsatisfies [package provide Tcl] 9.0-]} {
     package ifneeded @PACKAGE_NAME@ @PACKAGE_VERSION@ [list apply {{dir} {
 	# Load library
-	load [file join $dir @PKG_LIB_FILE9@] [string totitle @PACKAGE_NAME@]
+	if {[string tolower [file extension @PKG_LIB_FILE9@]] in [list .dll .dylib .so]} {
+	    # Load dynamic library
+	    load [file join $dir @PKG_LIB_FILE9@] [string totitle @PACKAGE_NAME@]
+	} else {
+	    # Static library
+	    load {} [string totitle @PACKAGE_NAME@]
+	}

 	# Source init file
 	set initScript [file join $dir @PACKAGE_NAME@.tcl]
