--- tcl9.0.3.orig/unix/Makefile.in	2025-11-11 12:22:57.000000000 -0400
+++ tcl9.0.3/unix/Makefile.in	2026-04-10 13:56:59.302753272 -0300
@@ -2048,7 +2048,6 @@
 # cannot use absolute paths due to issues in nested configure when path to
 # build dir contains spaces).
 PKG_DIR			= ./pkgs
-PKG8_DIR			= ./pkgs8
 
 configure-packages:
 	@for i in $(PKGS_DIR)/*; do \
@@ -2056,14 +2055,6 @@
 		if [ -x $$i/configure ] ; then \
 		    pkg=`basename $$i`; \
 		    echo "Configuring package '$$pkg'"; \
-		    mkdir -p $(PKG8_DIR)/$$pkg; \
-		    if [ ! -f $(PKG8_DIR)/$$pkg/Makefile ] ; then \
-			( cd $(PKG8_DIR)/$$pkg; \
-			  $$i/configure --with-tcl8 --with-tcl=../.. \
-			      --with-tclinclude=$(GENERIC_DIR) \
-			      $(PKG_CFG_ARGS) --libdir=$(PACKAGE_DIR) \
-			      --enable-shared; ) || exit $$?; \
-		    fi; \
 		    mkdir -p $(PKG_DIR)/$$pkg; \
 		    if [ ! -f $(PKG_DIR)/$$pkg/Makefile ] ; then \
 			( cd $(PKG_DIR)/$$pkg; \
@@ -2080,10 +2071,6 @@
 	@for i in $(PKGS_DIR)/*; do \
 	    if [ -d $$i ] ; then \
 		pkg=`basename $$i`; \
-		if [ -f $(PKG8_DIR)/$$pkg/Makefile ] ; then \
-		    echo "Building package '$$pkg' for Tcl 8"; \
-		    ( cd $(PKG8_DIR)/$$pkg; $(MAKE); ) || exit $$?; \
-		fi; \
 		if [ -f $(PKG_DIR)/$$pkg/Makefile ] ; then \
 		    echo "Building package '$$pkg'"; \
 		    ( cd $(PKG_DIR)/$$pkg; $(MAKE); ) || exit $$?; \
@@ -2095,11 +2082,6 @@
 	@for i in $(PKGS_DIR)/*; do \
 	    if [ -d $$i ] ; then \
 		pkg=`basename $$i`; \
-		if [ -f $(PKG8_DIR)/$$pkg/Makefile ] ; then \
-		    echo "Installing package '$$pkg' for Tcl 8"; \
-		    ( cd $(PKG8_DIR)/$$pkg; $(MAKE) install \
-			  "DESTDIR=$(INSTALL_ROOT)"; ) || exit $$?; \
-		fi; \
 		if [ -f $(PKG_DIR)/$$pkg/Makefile ] ; then \
 		    echo "Installing package '$$pkg'"; \
 		    ( cd $(PKG_DIR)/$$pkg; $(MAKE) install \
@@ -2127,9 +2109,6 @@
 	@for i in $(PKGS_DIR)/*; do \
 	    if [ -d $$i ] ; then \
 		pkg=`basename $$i`; \
-		if [ -f $(PKG8_DIR)/$$pkg/Makefile ] ; then \
-		    ( cd $(PKG8_DIR)/$$pkg; $(MAKE) clean; ) \
-		fi; \
 		if [ -f $(PKG_DIR)/$$pkg/Makefile ] ; then \
 		    ( cd $(PKG_DIR)/$$pkg; $(MAKE) clean; ) \
 		fi; \
@@ -2140,17 +2119,12 @@
 	@for i in $(PKGS_DIR)/*; do \
 	    if [ -d $$i ] ; then \
 		pkg=`basename $$i`; \
-		if [ -f $(PKG8_DIR)/$$pkg/Makefile ] ; then \
-		    ( cd $(PKG8_DIR)/$$pkg; $(MAKE) distclean; ) \
-		fi; \
-		rm -rf $(PKG8_DIR)/$$pkg; \
 		if [ -f $(PKG_DIR)/$$pkg/Makefile ] ; then \
 		    ( cd $(PKG_DIR)/$$pkg; $(MAKE) distclean; ) \
 		fi; \
 		rm -rf $(PKG_DIR)/$$pkg; \
 	    fi; \
 	done; \
-	rm -rf $(PKG8_DIR)
 	rm -rf $(PKG_DIR)
 
 dist-packages: configure-packages
--- tcl9.0.3.orig/win/Makefile.in	2025-11-11 12:22:58.000000000 -0400
+++ tcl9.0.3/win/Makefile.in	2026-04-10 13:57:34.059162777 -0300
@@ -541,7 +541,7 @@
 
 binaries: $(TCL_STUB_LIB_FILE) @LIBRARIES@ winextensions ${TCL_ZIP_FILE} $(TCLSH)
 
-winextensions: ${DDE_DLL_FILE} ${REG_DLL_FILE} ${DDE_DLL_FILE8} ${REG_DLL_FILE8}
+winextensions: ${DDE_DLL_FILE} ${REG_DLL_FILE}
 
 libraries:
 
@@ -616,14 +616,6 @@
 	@MAKE_DLL@ ${REG_OBJS} $(TCL_STUB_LIB_FILE) $(SHLIB_LD_LIBS)
 	$(COPY) tclsh.exe.manifest ${REG_DLL_FILE}.manifest
 
-${DDE_DLL_FILE8}: ${TCL_STUB_LIB_FILE} tcl8WinDde.$(OBJEXT)
-	@MAKE_DLL@ tcl8WinDde.$(OBJEXT) $(TCL_STUB_LIB_FILE) $(SHLIB_LD_LIBS)
-	$(COPY) tclsh.exe.manifest ${DDE_DLL_FILE8}.manifest
-
-${REG_DLL_FILE8}: ${TCL_STUB_LIB_FILE} tcl8WinReg.$(OBJEXT)
-	@MAKE_DLL@ -DTCL_MAJOR_VERSION=8 tcl8WinReg.$(OBJEXT) $(TCL_STUB_LIB_FILE) $(SHLIB_LD_LIBS)
-	$(COPY) tclsh.exe.manifest ${REG_DLL_FILE8}.manifest
-
 ${TEST_DLL_FILE}: ${TCL_STUB_LIB_FILE} ${TCLTEST_OBJS}
 	@$(RM) ${TEST_DLL_FILE} ${TEST_LIB_FILE}
 	@MAKE_DLL@ ${TCLTEST_OBJS} $(TCL_STUB_LIB_FILE) $(SHLIB_LD_LIBS)
@@ -680,15 +672,9 @@
 tclWinReg.${OBJEXT}: tclWinReg.c
 	$(CC) -c $(CC_SWITCHES) $(EXTFLAGS) @DEPARG@ $(CC_OBJNAME)
 
-tcl8WinReg.${OBJEXT}: tclWinReg.c
-	$(CC)  -o $@ -c $(CC_SWITCHES) -DTCL_MAJOR_VERSION=8 $(EXTFLAGS) @DEPARG@ $(CC_OBJNAME)
-
 tclWinDde.${OBJEXT}: tclWinDde.c
 	$(CC) -c $(CC_SWITCHES) $(EXTFLAGS) @DEPARG@ $(CC_OBJNAME)
 
-tcl8WinDde.${OBJEXT}: tclWinDde.c
-	$(CC) -o $@ -c $(CC_SWITCHES) -DTCL_MAJOR_VERSION=8 $(EXTFLAGS) @DEPARG@ $(CC_OBJNAME)
-
 tclAppInit.${OBJEXT}: tclAppInit.c
 	$(CC) -c $(CC_SWITCHES) $(EXTFLAGS) -DUNICODE -D_UNICODE @DEPARG@ $(CC_OBJNAME)
 
@@ -885,10 +871,6 @@
 	    $(COPY) $(ROOT_DIR)/library/dde/pkgIndex.tcl \
 		"$(LIB_INSTALL_DIR)/dde${DDEDOTVER}"; \
 	    fi
-	@if [ -f $(DDE_DLL_FILE8) ]; then \
-	    echo Installing $(DDE_DLL_FILE8); \
-	    $(COPY) $(DDE_DLL_FILE8) "$(LIB_INSTALL_DIR)/dde${DDEDOTVER}"; \
-	    fi
 	@if [ -f $(DDE_LIB_FILE) ]; then \
 	    echo Installing $(DDE_LIB_FILE); \
 	    $(COPY) $(DDE_LIB_FILE) "$(LIB_INSTALL_DIR)/dde${DDEDOTVER}"; \
@@ -899,10 +881,6 @@
 	    $(COPY) $(ROOT_DIR)/library/registry/pkgIndex.tcl \
 		"$(LIB_INSTALL_DIR)/registry${REGDOTVER}"; \
 	    fi
-	@if [ -f $(REG_DLL_FILE8) ]; then \
-	    echo Installing $(REG_DLL_FILE8); \
-	    $(COPY) $(REG_DLL_FILE8) "$(LIB_INSTALL_DIR)/registry${REGDOTVER}"; \
-	    fi
 	@if [ -f $(REG_LIB_FILE) ]; then \
 	    echo Installing $(REG_LIB_FILE); \
 	    $(COPY) $(REG_LIB_FILE) "$(LIB_INSTALL_DIR)/registry${REGDOTVER}"; \
