diff -ruN A/cmpRPkg.c B/cmpRPkg.c
--- A/cmpRPkg.c	2024-01-01 00:00:00.000000000 +0000
+++ B/cmpRPkg.c	2026-04-05 00:00:00.000000000 +0000
@@ -89,10 +89,10 @@
 {
 #ifdef USE_TCL_STUBS
-    if (!Tcl_InitStubs(interp, TCL_VERSION, 1))
+    if (!Tcl_InitStubs(interp, TCL_VERSION, 0))
     {
         return TCL_ERROR;
     }
 #else
-    if (Tcl_PkgRequire(interp, "Tcl", TCL_VERSION, 1) == NULL)
+    if (Tcl_PkgRequire(interp, "Tcl", TCL_VERSION, 0) == NULL)
     {
         return TCL_ERROR;
