diff -ruN A/src/library.c B/src/library.c
--- A/src/library.c	2026-07-31 17:05:24.177942523 -0300
+++ B/src/library.c	2026-07-31 17:05:38.555199706 -0300
@@ -268,9 +268,14 @@
 
     if (length > 0) {
         cJSON *root_structure = cJSON_ParseWithLength(json, length);
-        Tcl_Obj *resultPtr = tjson_TreeToTyped(interp, root_structure);
-        cJSON_Delete(root_structure);
-        Tcl_SetObjResult(interp, resultPtr);
+        if (root_structure) {
+            Tcl_Obj *resultPtr = tjson_TreeToTyped(interp, root_structure);
+            cJSON_Delete(root_structure);
+            Tcl_SetObjResult(interp, resultPtr);
+        } else {
+            Tcl_SetObjResult(interp, Tcl_NewStringObj("invalid json", -1));
+            return TCL_ERROR;
+        }
     }
 
     return TCL_OK;
