Index: unix/Makefile.in
==================================================================
--- A/unix/Makefile.in
+++ B/unix/Makefile.in
@@ -394,11 +394,12 @@
 	tkVisual.o tkWindow.o
 
 TTK_OBJS = \
 	ttkBlink.o ttkButton.o ttkCache.o ttkClamTheme.o ttkClassicTheme.o \
 	ttkDefaultTheme.o ttkElements.o ttkEntry.o ttkFrame.o ttkImage.o \
-	ttkInit.o ttkLabel.o ttkLayout.o ttkManager.o ttkNotebook.o \
+	ttkInit.o ttkLabel.o ttkLayout.o ttkManager.o ttkNodeCache.o \
+	ttkNotebook.o \
 	ttkPanedwindow.o ttkProgress.o ttkScale.o ttkScrollbar.o ttkScroll.o \
 	ttkSeparator.o ttkSquare.o ttkState.o \
 	ttkTagSet.o ttkTheme.o ttkTrace.o ttkTrack.o ttkTreeview.o \
 	ttkWidget.o ttkStubInit.o
 
@@ -497,10 +498,11 @@
 	$(TTK_DIR)/ttkImage.c \
 	$(TTK_DIR)/ttkInit.c \
 	$(TTK_DIR)/ttkLabel.c \
 	$(TTK_DIR)/ttkLayout.c \
 	$(TTK_DIR)/ttkManager.c \
+	$(TTK_DIR)/ttkNodeCache.c \
 	$(TTK_DIR)/ttkNotebook.c \
 	$(TTK_DIR)/ttkPanedwindow.c \
 	$(TTK_DIR)/ttkProgress.c \
 	$(TTK_DIR)/ttkScale.c \
 	$(TTK_DIR)/ttkScrollbar.c \
@@ -1542,10 +1544,13 @@
 	$(CC) -c $(CC_SWITCHES) $(TTK_DIR)/ttkLabel.c
 
 ttkLayout.o: $(TTK_DIR)/ttkLayout.c
 	$(CC) -c $(CC_SWITCHES) $(TTK_DIR)/ttkLayout.c
 
+ttkNodeCache.o: $(TTK_DIR)/ttkNodeCache.c
+	$(CC) -c $(CC_SWITCHES) $(TTK_DIR)/ttkNodeCache.c
+
 ttkManager.o: $(TTK_DIR)/ttkManager.c
 	$(CC) -c $(CC_SWITCHES) $(TTK_DIR)/ttkManager.c
 
 ttkNotebook.o: $(TTK_DIR)/ttkNotebook.c
 	$(CC) -c $(CC_SWITCHES) $(TTK_DIR)/ttkNotebook.c

Index: win/Makefile.in
==================================================================
--- A/win/Makefile.in
+++ B/win/Makefile.in
@@ -421,10 +421,11 @@
 	ttkImage.$(OBJEXT) \
 	ttkInit.$(OBJEXT) \
 	ttkLabel.$(OBJEXT) \
 	ttkLayout.$(OBJEXT) \
 	ttkManager.$(OBJEXT) \
+	ttkNodeCache.$(OBJEXT) \
 	ttkNotebook.$(OBJEXT) \
 	ttkPanedwindow.$(OBJEXT) \
 	ttkProgress.$(OBJEXT) \
 	ttkScale.$(OBJEXT) \
 	ttkScrollbar.$(OBJEXT) \

Index: win/makefile.vc
==================================================================
--- A/win/makefile.vc
+++ B/win/makefile.vc
@@ -287,10 +287,11 @@
 	$(TMP_DIR)\ttkImage.obj \
 	$(TMP_DIR)\ttkInit.obj \
 	$(TMP_DIR)\ttkLabel.obj \
 	$(TMP_DIR)\ttkLayout.obj \
 	$(TMP_DIR)\ttkManager.obj \
+	$(TMP_DIR)\ttkNodeCache.obj \
 	$(TMP_DIR)\ttkNotebook.obj \
 	$(TMP_DIR)\ttkPanedwindow.obj \
 	$(TMP_DIR)\ttkProgress.obj \
 	$(TMP_DIR)\ttkScale.obj \
 	$(TMP_DIR)\ttkScrollbar.obj \

