diff --git a/configure.ac b/configure.ac
index 10a6e8464f9..1dca82b87b0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -180,7 +180,7 @@ AC_MSG_CHECKING([for Python interpreter freezing])
 AC_MSG_RESULT([$PYTHON_FOR_FREEZE])
 AC_SUBST([PYTHON_FOR_FREEZE])
 
-AS_VAR_IF([cross_compiling], [yes],
+AS_IF([test "_${cross_compiling}" = "_yes" || test -n "$with_build_python"],
   [
     dnl external build Python, freezing depends on Programs/_freeze_module.py
     FREEZE_MODULE_BOOTSTRAP='$(PYTHON_FOR_FREEZE) $(srcdir)/Programs/_freeze_module.py'
