--- trf2.1.4.bak/generic/sha/sha.h	Thu Jun 18 01:45:22 2009
+++ trf2.1.4/trf2.1.4/generic/sha/sha.h	Mon Jan  8 23:54:45 2024
@@ -10,16 +10,16 @@
 
 #ifndef _WIN32
 typedef unsigned char BYTE;
-#endif
-#if defined(__alpha) || defined(__LP64__)
+#  if defined(__alpha) || defined(__LP64__)
 typedef unsigned int  UINT32;
-#else
-#ifndef UINT32
-#ifdef _WIN32
-#	pragma warning ( disable : 4142 )
-#endif
+#  else
+#    ifndef UINT32
+#      ifdef _WIN32
+#	      pragma warning ( disable : 4142 )
+#      endif
 typedef unsigned long UINT32;
-#endif
+#    endif
+#  endif
 #endif
 
 
